darwin/PlatformHelpers.c

```diff
@@ -86,3 +86,42 @@ bool Platform_isRunningTranslated(void) {
    }
    return ret;
 }
+
+void Platform_calculateNanosecondsPerMachTick(uint64_t* numer, uint64_t* denom) {
+   // Check if we can determine the timebase used on this system.
+   // If the API is unavailable assume we get our timebase in nanoseconds.
+#ifdef HAVE_MACH_TIMEBASE_INFO
+   mach_timebase_info_data_t info;
+   mach_timebase_info(&info);
```

Comment:
This probably should check for errors, AFAICS [here](https://developer.apple.com/documentation/driverkit/3433733-mach_timebase_info).. Falling back to some "sane" timebase on error, won't hurt. Otherwise the timebase is undefined.